### ISSUE TITLE
Learning rate about the hyperparameter beta

### ISSUE THREAD

**Issue by wensihan** [sniklaus/softmax-splatting#4]
Hello, niklaus:

When I try to add the beta to the network, after training for a few epoches, it turn out to be None, the learning rate was initially set to 0.001, is it convenient for you to give me some advice of this part?

Looking forward for your reply~

Wen

**Comment by sniklaus** [comment]
Thank you for your interest in our work!

In my opinion, the learning rate stated in a paper is pretty meaningless. There are many factors at play, which end up making it necessary to find a suitable learning rate yourself in cases where the authors did not publish their full training pipeline (my apologies that I am not allowed to do so).

Consider the loss reduction for example. One could use a sum or an average, and the choice will influence the learning rate (granted, the definition of the loss should already be clear about this aspect).

Anyways, I am not surprised that the learning rate that worked well in our training setup does not yield ideal results on your end. I would thus just try a few different ones, like 0.01 or 0.0001, and see what works best in your training regime.

**Comment by wensihan** [comment]
Niklaus,

Thank you for your advice, I will do more try about the training, or change the current loss function.

Wen

**Comment by asheroin** [comment]
> Niklaus,
> 
> Thank you for your advice, I will do more try about the training, or change the current loss function.
> 
> Wen

Intresting about your repeoducing works, would you please leave an email for further disscussion?

**Comment by wensihan** [comment]
> > Niklaus,
> > Thank you for your advice, I will do more try about the training, or change the current loss function.
> > Wen
> 
> Intresting about your repeoducing works, would you please leave an email for further disscussion?

Sorry for late reply, chengyissing@163.com

**Comment by wensihan** [comment]
> Thank you for your interest in our work!
> 
> In my opinion, the learning rate stated in a paper is pretty meaningless. There are many factors at play, which end up making it necessary to find a suitable learning rate yourself in cases where the authors did not publish their full training pipeline (my apologies that I am not allowed to do so).
> 
> Consider the loss reduction for example. One could use a sum or an average, and the choice will influence the learning rate (granted, the definition of the loss should already be clear about this aspect).
> 
> Anyways, I am not surprised that the learning rate that worked well in our training setup does not yield ideal results on your end. I would thus just try a few different ones, like 0.01 or 0.0001, and see what works best in your training regime.

I have find out the reason for Nan loss. As your paper illustrate, I use an u-net to refine the importance metric, while the last Relu activation function of u-net lead to Nan when feed into exp(). I am wondering what kind of method have you used to control the output range of importance metric? Sigmoid or other method?

Looking forward for you reply~

**Comment by sniklaus** [comment]
The output layer in this U-Net is a convolution without a subsequent activation function. Note that, as shown in Table 1 of our paper, fine-tuning Z does not have a huge impact on the performance. Our motivation behind fine-tuning Z was to show that all inputs to the softmax splatting can be optimized (the features, the flow, the metric). So you could try not using this U-Net for now. Instead, something like the following should do.

```
class Metric(torch.nn.Module):
	def __init__(self):
		super(Metric, self).__init__()

		self.paramScale = torch.nn.Parameter(-torch.ones(1, 1, 1, 1))
	# end

	def forward(self, tenFirst, tenSecond, tenFlow):
		return self.paramScale * torch.nn.functional.l1_loss(input=tenFirst, target=backwarp(tenSecond, tenFlow)).mean(1, True)
	# end
# end
```

**Comment by wensihan** [comment]
@sniklaus Thanks for your explaination, I will do some trials as you suggested~
